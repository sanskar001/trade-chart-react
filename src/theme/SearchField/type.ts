type ChangeHandlerType = React.ChangeEventHandler<HTMLInputElement>;

type PropsType =
  | string
  | boolean
  | undefined
  | ChangeHandlerType
  | React.CSSProperties;

export interface SearchFieldProps {
  value: string;
  onChange: ChangeHandlerType;
  className?: string;
  style?: React.CSSProperties;
  autoFocus?: boolean;
  disabled?: boolean;
  placeholder?: string;
  [key: string]: PropsType;
}
