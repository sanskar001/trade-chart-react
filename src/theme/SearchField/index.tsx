import React from "react";
import SearchIcon from "../../components/SVG/SearchIcon";
import { formatClass } from "../../util/formatClass";
import { SearchFieldProps } from "./type";

const SearchField: React.FC<SearchFieldProps> = ({
  value,
  onChange,
  className = "",
  style,
  autoFocus = true,
  disabled = false,
  placeholder = "Search",
  ...props
}) => {
  return (
    <div className={formatClass(`search-field ${className}`)} style={style}>
      <label htmlFor="search">
        <SearchIcon />
      </label>
      <input
        id="search"
        type="text"
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        autoComplete="off"
        autoFocus={autoFocus}
        disabled={disabled}
        {...props}
      />
    </div>
  );
};

export default SearchField;
