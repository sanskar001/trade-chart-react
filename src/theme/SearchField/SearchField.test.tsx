import renderer from "react-test-renderer";
import SearchField from ".";

const changeHandler = jest.fn();

it("renders correctly when there is default searchfield", () => {
  const tree = renderer
    .create(<SearchField value="APPLE" onChange={changeHandler} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is searchfield with custom class", () => {
  const tree = renderer
    .create(
      <SearchField
        value="APPLE"
        onChange={changeHandler}
        className="bg-green-200"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is searchfield with custom style", () => {
  const tree = renderer
    .create(
      <SearchField
        value="APPLE"
        onChange={changeHandler}
        style={{ backgroundColor: "lightcyan" }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is autoFocus searchfield", () => {
  const tree = renderer
    .create(<SearchField value="APPLE" onChange={changeHandler} autoFocus />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is searchfield other attributes", () => {
  const tree = renderer
    .create(
      <SearchField
        value="APPLE"
        onChange={changeHandler}
        name="search symbol"
        placeholder="Search Symbol"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
