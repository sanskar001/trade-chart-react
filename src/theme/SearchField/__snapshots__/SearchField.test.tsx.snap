// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is autoFocus searchfield 1`] = `
<div
  className="search-field"
>
  <label
    htmlFor="search"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 15.707H16V22.707H15z"
        fill="#A3A6AF"
        transform="rotate(-45 15 15.707)"
      />
      <circle
        cx="10.5"
        cy="10.5"
        r="7"
        stroke="#A3A6AF"
      />
    </svg>
  </label>
  <input
    autoComplete="off"
    autoFocus={true}
    disabled={false}
    id="search"
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value="APPLE"
  />
</div>
`;

exports[`renders correctly when there is default searchfield 1`] = `
<div
  className="search-field"
>
  <label
    htmlFor="search"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 15.707H16V22.707H15z"
        fill="#A3A6AF"
        transform="rotate(-45 15 15.707)"
      />
      <circle
        cx="10.5"
        cy="10.5"
        r="7"
        stroke="#A3A6AF"
      />
    </svg>
  </label>
  <input
    autoComplete="off"
    autoFocus={false}
    disabled={false}
    id="search"
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value="APPLE"
  />
</div>
`;

exports[`renders correctly when there is searchfield other attributes 1`] = `
<div
  className="search-field"
>
  <label
    htmlFor="search"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 15.707H16V22.707H15z"
        fill="#A3A6AF"
        transform="rotate(-45 15 15.707)"
      />
      <circle
        cx="10.5"
        cy="10.5"
        r="7"
        stroke="#A3A6AF"
      />
    </svg>
  </label>
  <input
    autoComplete="off"
    autoFocus={false}
    disabled={false}
    id="search"
    name="search symbol"
    onChange={[MockFunction]}
    placeholder="Search Symbol"
    type="text"
    value="APPLE"
  />
</div>
`;

exports[`renders correctly when there is searchfield with custom class 1`] = `
<div
  className="search-field bg-green-200"
>
  <label
    htmlFor="search"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 15.707H16V22.707H15z"
        fill="#A3A6AF"
        transform="rotate(-45 15 15.707)"
      />
      <circle
        cx="10.5"
        cy="10.5"
        r="7"
        stroke="#A3A6AF"
      />
    </svg>
  </label>
  <input
    autoComplete="off"
    autoFocus={false}
    disabled={false}
    id="search"
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value="APPLE"
  />
</div>
`;

exports[`renders correctly when there is searchfield with custom style 1`] = `
<div
  className="search-field"
  style={
    Object {
      "backgroundColor": "lightcyan",
    }
  }
>
  <label
    htmlFor="search"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 15.707H16V22.707H15z"
        fill="#A3A6AF"
        transform="rotate(-45 15 15.707)"
      />
      <circle
        cx="10.5"
        cy="10.5"
        r="7"
        stroke="#A3A6AF"
      />
    </svg>
  </label>
  <input
    autoComplete="off"
    autoFocus={false}
    disabled={false}
    id="search"
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value="APPLE"
  />
</div>
`;
